src/app/predict/uploadMedia/page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import PredictionLayout from "@/components/PredictionLayout";
+import SquareIconButton from "@/components/SquareIconButton";
+import { Model } from "@/types/model";
+import { useRef, useState } from "react";
+interface MediaState {
+  typeOfMedia: "video" | "image" | null;
+}
+
+export default function RealtimePredictionPage() {
+  const [mediaUpload, setMediaUpload] = useState<boolean>(false);
```

Comment:
You can remove this in favor of allowing mediaList to be null <string[] | null>. Check instead whether mediaList is null or not. The less states they are should reduce chance of error. See selectedModel for an example.